orttraining/orttraining/core/optimizer/memory_optimizer/common.h

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "core/common/common.h"
+#include "core/common/logging/logging.h"
+#include "core/common/inlined_containers.h"
+#include "core/common/const_pointer_container.h"
+#include "core/common/status.h"
+#include "core/graph/basic_types.h"
+#include "core/graph/constants.h"
+#include "core/graph/graph_nodes.h"
+#include "core/graph/graph.h"
```

Comment:
This looks like a pre-compiled header. At the same time lots of code that does not need it, would be including it. Please, move out things that are related to the implementation, not interfaces.